opentracing-proxy/opentracing-proxy/src/main/java/org/zalando/opentracing/proxy/plugin/LogCorrelation.java

```diff
@@ -75,9 +78,16 @@ public LogCorrelation withBaggage(final Map<String, String> mapping) {
 
     @Override
     public void onActivated(final Scope scope, final Span span) {
+        if (existingContextStack.get() == null)
+            existingContextStack.set(new Stack<>());
+        Map<String, String> contextMap = new HashMap<>();
         seeds.forEach((seed, contextKey) ->
-                Optional.ofNullable(seed.valueOf(span)).ifPresent(value ->
-                        MDC.put(contextKey, value)));
+                Optional.ofNullable(seed.valueOf(span)).ifPresent(value -> {
+                    Optional.ofNullable(MDC.get(contextKey)).ifPresent(x -> contextMap.put(contextKey, x));
+                    MDC.put(contextKey, value);
+                }));
+        if (!contextMap.isEmpty())
```

Comment:
This is just an optimization, right?

Reply:
yes, in case there isn't anything to save in context, on closing we can remove the `ThreadLocal`